src/Abstract/Interpreter/Caching.hs

```diff
@@ -119,6 +120,7 @@ fixCache eval e = do
   pairs <- mlfp mempty (\ dollar -> do
     putCache (mempty :: Cache l t v)
     putStore store
+    reset 0
```

Comment:
We reset the fresh variable counter on each iteration to avoid needing to normalize here (which would specialize the caching interpreter to `Type` unnecessarily).